src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -1398,10 +1398,31 @@ private void visitVarDeclaration(String name, JSType type) {
     }
 
     private void visitTemplateTypes(ObjectType type) {
-      visitTemplateTypes(type, Collections.<String>emptySet());
+      visitTemplateTypes(type, Collections.<String>emptySet(), true);
     }
 
-    private void visitTemplateTypes(ObjectType type, Set<String> alreadyEmittedTemplateType) {
+    /**
+     * Emits template types for a given type. For example for
+     *
+     * <pre>T<A,B></pre>
+     *
+     * , this method will emit
+     *
+     * <pre><A,B></pre>
+     *
+     * .
+     *
+     * @param type the type in question
+     * @param alreadyEmittedTemplateType in certain scopes some template types will be already
+     *     emitted and have to be skipped.
+     * @param isDeclaration this method will be called when defining new template types, like
```

Comment:
`this method ...` is a bit misleading as the doc string start, isn't it? maybe `isDeclaration {@pre true} if this type declares the template types, {@pre false} if it instantiates a generic type. In the former case, Clutz emits defaults for the template parameters.`